client/src/components/Chat/ChatDetails/ChatMessages/ChatMessages.jsx

```diff
@@ -1,35 +1,86 @@
-import React from 'react'
+import React, { useEffect, useRef } from 'react'
 import PropTypes from 'prop-types'
 import classnames from 'classnames'
 
 import ChatMessage from '../ChatMessage/ChatMessage'
+import InfiniteScroll from '../../../InfiniteScroll/InfiniteScroll'
 
 import useStyles from './chatMessagesStyles'
 
-const ChatMessages = ({ authUser, messages, className }) => {
+const ChatMessages = ({
+  authUser,
+  messages,
+  className,
+  messagesLoading,
+  loadContentHandler,
+  ownMessageSend,
+  isLastPageInChat }) => {
   const classes = useStyles()
+  let isInitialScroll = useRef(true)
+  const scrollToBottom = () => {
+    ChatMessages.messagesEnd.scrollIntoView({ behavior: 'smooth' })
+  }
+  useEffect(() => {
+    if (isInitialScroll.current || ownMessageSend) {
+      scrollToBottom()
+    } else {
+      setTimeout(() => { ChatMessages.messagesStart.scrollIntoView({ behavior: 'smooth' }) }, 500)
+    }
+    setTimeout(() => { isInitialScroll.current = false }, 500)
+  })
 
   return (
-    <div className={classnames(classes.root, className)}>
-      <div className={classes.inner}>
-        {messages.map(message => {
-          return (
-            <ChatMessage
-              key={message.id}
-              message={message}
-              authUser={authUser}
-              />
-          )
-        })}
+    <InfiniteScroll
+      isReverseDirection
+      contentIsLoading={messagesLoading}
+      contentArr={messages}
+      loadContentHandler={loadContentHandler}
+      size={7}
+      throttleDelay={1000}
+      isLastPage={isLastPageInChat}
+      scrollContainerStyles={{
+        height: '80vh',
```

Comment:
On homepage posts container is cut. Check please, maybe problem is with this property
<img width="802" alt="image" src="https://user-images.githubusercontent.com/48453944/73700167-778b4c80-46ee-11ea-8bfa-c41ff62c0ecf.png">

Reply:
Thanks, forgot to remove  height: '80vh' from scrollContainerStyles